core/vpn/src/network.rs

```diff
@@ -820,4 +843,40 @@ mod tests {
 
         Ok(())
     }
+
+    #[actix_rt::test]
+    async fn create_remove_2_networks() -> anyhow::Result<()> {
+        let node_id = NodeId::default();
+
+        let mut supervisor = VpnSupervisor::default();
+        let network1 = supervisor
+            .create_network(
+                node_id,
+                NewNetwork {
+                    ip: "10.0.0.0".to_string(),
+                    mask: None,
+                    gateway: None,
+                },
+            )
+            .await?;
+        let network2 = supervisor
+            .create_network(
+                node_id,
+                NewNetwork {
+                    ip: "10.0.0.0".to_string(),
```

Comment:
I had a moment today during investigation that I thought the same. But in our virtual network we have `network_id` that uniquely identifies the network.

If we would forbid having the same addressing than it would be big inconvenient for `dapp-portal` for example. In `golem-descriptor` you have static ip addresses. If you would like to run the same application many times, than you would need to change addressing in this descriptor dynamically